Shared/ReflectionHelper.cs

```diff
@@ -1891,7 +1891,25 @@ internal class Time
         /// <returns></returns>
         public static TimeSpan FromSeconds(double seconds)
         {
-            return TimeSpan.FromTicks((long)(seconds * 1E7));
+            try
+            {
+                checked
+                {
+                    // if the multiplication here creates a number greater than
+                    // long.Max (or overflows negatively), we just return
+                    // TimeSpan.MaxValue/MinValue.
+                    long ticks = (long)(seconds * 1e7);
+                    return TimeSpan.FromTicks(ticks);
```

Comment:
Why not just use `TimeSpan.FromSeconds(seconds)`? That variant also throws an overflow exception. It has the same behavior as this checked computation, except it throws an `ArgumentException` if seconds is double.NaN. Should we handle the NaN case?

Reply:
That would be awesome, but last I checked, TimeSpan.FromSeconds rounds to nearest whole millisecond, so we don't use that.